scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,506 @@
+        opt.setArgs(0);
+        opt.setRequired(false);
+        optionGroup.addOption(opt);
+
+        opt = new Option(UPDATE_OPTION, "update", true,
+                "Action to update an existing user. Updating a user means to change the user's password or group membership.");
+        opt.setArgName("UPDATE");
+        opt.setArgs(0);
+        opt.setRequired(false);
+        optionGroup.addOption(opt);
+
+        opt = new Option(DELETE_OPTION, "delete", true,
+                "Action to delete an existing user");
+        opt.setArgName("DELETE");
+        opt.setArgs(0);
+        opt.setRequired(false);
+        optionGroup.addOption(opt);
+        options.addOptionGroup(optionGroup);
+
+        opt = new Option(LOGIN_OPTION, "login", true,
+                "Generate credentials for this specific user, will be asked interactively if not specified");
+        opt.setArgName("LOGIN");
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(PWD_OPTION, "password", true,
+                "Password of the user, if the user is created or updated, will be asked interactively if not specified");
+        opt.setArgName("PWD");
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(GROUPS_OPTION, "groups", true,
+                "A comma-separated list of groups the user must be member of. Can be used when the user is created or updated");
+        opt.setArgName("GROUPS");
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        optionGroup.setRequired(false);
+        opt = new Option(KEY_OPTION, "key", true, "Public key path on the local filesystem [default:" +
+                getPublicKeyFilePath() + "]");
+        opt.setArgName("KEY");
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(LOGINFILE_OPTION, "loginfile", true,
+                "Path to the login file in use [default:" + loginFilePath + "]");
+        opt.setArgName("LOGINFILE");
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option(GROUPFILE_OPTION, "groupfile", true,
+                "Path to the group file in use [default:" + groupFilePath + "]");
+        opt.setArgName("GROUPFILE");
+        opt.setArgs(1);
+        opt.setRequired(false);
+        options.addOption(opt);
+
+
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = null;
+        try {
+            cmd = parser.parse(options, args);
+        } catch (Exception e) {
+            exitWithErrorMessage(newline + e.getMessage() + newline, "type -h or --help to display help screen", null);
+        }
+
+        if (cmd.hasOption("help")) {
+            displayHelp(options);
+            return;
+        }
+
+        Boolean create = cmd.hasOption("create");
```

Comment:
you could pass **cmd** to the Action enum constructor and them use only the Action enum in this class. otherwise you have 2 fields representing the same concept and can lead to bugs